src/main/scala/edu/berkeley/cs/rise/opaque/Utils.scala

```diff
@@ -291,21 +291,25 @@ object Utils extends Logging {
   val numAttested: LongAccumulator = new LongAccumulator
   var loop: Boolean = true
 
-  def initSQLContext(sqlContext: SQLContext): Unit = {
+  def initOpaqueSQL(spark: SparkSession): Unit = {
+    val sqlContext = spark.sqlContext
     sqlContext.experimental.extraOptimizations =
       (Seq(EncryptLocalRelation, ConvertToOpaqueOperators) ++
         sqlContext.experimental.extraOptimizations)
     sqlContext.experimental.extraStrategies = (Seq(OpaqueOperators) ++
       sqlContext.experimental.extraStrategies)
 
+    // TODO: replace this with loading in a key from a file specified in conf.
+    sharedKey = Some(Array.fill[Byte](GCM_KEY_LENGTH)(1))
+
```

Comment:
Does this mean the `sharedKey` will always need to be set with a user-specified key in the future? How can users run Opaque in test mode?

Reply:
For test mode, there is still the `spark.opaque.testing.enableSharedKey` config.

Reply:
Per our offline conversation: would need to add a "read key from file" functionality here; also need to put that in the documentation.